### DISCUSSION TITLE
How can I read the configurations?

### DISCUSSION THREAD

**Discussion by ghost** [symfony/symfony#47147] [Q&A] [answered]
Hello everybody, I'm new to Symfony, and I'm studying Symfony 6.1 on the official website.

But, I can not get how to read configuration,  from controller, service, ...

Should I use the dependency injector?
OR
Any approach?

Did the config file associate with service or controller or something esle using some rules?

HELP! Thanks!

**Comment by ahundiak** [comment, marked as answer]
This is not an uncommon request for new Symfony developers.  They see the config files and they want to access some of the data.  But no, that is not how Symfony works.  Symfony has what they term a compile stage.  During the compile stage all the config files are read and merged into sort of a master config and then used to generate services and what not.  The main result is a cached container which, if you are brave enough, you can explore by looking under the var/cache directory.  The actual config data is never stored as such.

If you want more details you can look at the docs for the config component as well as the container component.  It can also be useful to look in the DependencyInjection folder for some of the Symfony bundles where the compiling is done.  Not the easiest code to understand but you can get an idea of what is going on.

In general it is up to the bundle to expose config information typically via container parameters.  If the config information you want is not available then there usually a good reason.  And yes, you should be using Dependency Injection in your application.
